pints/tests/test_mcmc_relativistic.py

```diff
@@ -189,6 +191,91 @@ def test_other_setters(self):
         mcmc.set_hamiltonian_threshold(threshold2)
         self.assertEqual(mcmc.hamiltonian_threshold(), threshold2)
 
+    def test_momentum_logpdf(self):
+        # Test log pdf of momentum magnitude
+
+        x0 = np.array([2, 2, 2, 2, 2])
+        model = pints.RelativisticMCMC(x0)
+        m = 1.6
+        c = 3.4
+        n = len(x0)
+        model.set_mass(m)
+        model.set_speed_of_light(c)
+        model.ask()
+
+        mag = 1.7
+        f1 = model._momentum_logpdf(mag)
+        f2 = np.exp(-m * c**2 * np.sqrt(mag ** 2 / (m**2 * c**2) + 1)) \
+            * mag ** (n - 1)
+
+        self.assertAlmostEqual(f1, np.log(f2))
+
+    def test_calculate_momentum_distribution(self):
+        # Test calculation of inv cdf of momentum magnitude
+        x0 = np.array([2, 2, 2, 2, 2])
+        model = pints.RelativisticMCMC(x0)
+
+        # Choose values of m and c for which the distribution is known to be
+        # calculable without using logarithms
+        m = 1.6
+        c = 3.4
+        model.set_mass(m)
+        model.set_speed_of_light(c)
+        model.ask()
+
+        # Integrate the pdf to get normalizing constant
+        def pdf(u):
+            return np.exp(model._momentum_logpdf(u))
+
+        c = scipy.integrate.quad(pdf, 0, 100)[0]
+
+        # Integrate to get cumulative distribution function
+        integration_grid = np.arange(1e-6, 10, 1e-5)
+        cdf = scipy.integrate.cumtrapz(
+            1 / c * pdf(integration_grid), x=integration_grid)
```

Comment:
Do we think it's more stable (floating point wise I suppose) with the `np.logaddexp`?

Reply:
Yes, even for surprisingly moderate values of m and c, the distribution will underflow unless you use log. The test is a special case where I picked m and c to be nice so we could make this check.

Reply:
Awesome, would you mind say that in the comment, just in case later one might though this was a better method and try to 'improve' what you've got with `logaddexp` version.

Reply:
And maybe, even try to hard code a difficult one (with the choice of m and c) to test that it still works? And if one did try to 'improve' the method, it would at least show that it's somehow as good as yours (which this `cumtrapz` method fails). Only if there's an obvious choice, don't worry if that's too hard to find one.